tools/xtce_generation/xtce_generator_template.py

```diff
@@ -1,47 +1,104 @@
-"""
-Module for generating XTCE files for telemetry packets for a given instrument.
-
-This module provides functionality to generate XTCE files for telemetry packets
-for an instrument. It includes a `TelemetryGenerator` class for creating XTCE files
-based on packet definitions stored in an Excel file.
-"""
+"""XTCE generator script that uses CLI for generalized format."""
 
+import argparse
+import json
 from pathlib import Path
 
+import imap_data_access
+
 from tools.xtce_generation.telemetry_generator import TelemetryGenerator
```

Comment:
I am not sure that this will work as an external script, were you able to verify that this works from other directories? We aren't installing the tools folder, so this currently has to be referenced directly from the folder structure I think.

Reply:
@greglucas That makes sense! In the external libraries, `imap-data-access` is loaded in, and this is a line of code I totally stole from `cli.py`. Would that make this okay to keep as is?

I tried to download the imap-xtce in a different directory to test it, but was unable to. Whether or not this is user error or actual error I am unsure, so I reached out to the students slack-chat.

Reply:
Thanks for looking into this!

I think this is only an issue here because it is in the "tools" area, which is an extra portion and not something that gets added to the distribution when we type `pip install mypackage`. The way around this would be to inline `TelemetryGenerator`, or put this CLI work into the same file as that so there are no relative imports required.

I wouldn't bother with this. We can know about the limitation and this is an improvement 👍 . That can be added/updated as an extra feature later.